src/azure-cli-testsdk/azure/cli/testsdk/preparers.py

```diff
@@ -101,21 +111,26 @@ def __init__(self, name_prefix='clitest', sku='Standard_LRS', location='westus',
         self.parameter_name = parameter_name
         self.key = key
         self.dev_setting_name = os.environ.get(dev_setting_name, None)
+        self.subscription = subscription
 
     def create_resource(self, name, **kwargs):
         group = self._get_resource_group(**kwargs)
 
         if not self.dev_setting_name:
-            template = 'az storage account create -n {} -g {} -l {} --sku {} --kind {} --https-only --hns {}'
-            self.live_only_execute(self.cli_ctx, template.format(
-                name, group, self.location, self.sku, self.kind, self.hns))
+            cmd = 'az storage account create -n {} -g {} -l {} --sku {} --kind {} --https-only --hns {}'.format(
+                name, group, self.location, self.sku, self.kind, self.hns
+            )
+            if not self.subscription:
```

Comment:
if there way to adding `--subscription` logic into base class? otherwise each services involves cross subs scenarios need to take care of this by themselves. They need to find exmaple of how to do it themselves.

Reply:
Good idea.